contrib/slurm/slurmexecutor.py

```diff
@@ -0,0 +1,306 @@
+    srun_timelimit = f"{srun_timelimit_h}:{srun_timelimit_m}:{srun_timelimit_s}"
+
+    mem_per_cpu = int(memory / cpus / 1000000)
+
+    assert benchmark.config.scratchdir and os.path.exists(benchmark.config.scratchdir)
+
+    with tempfile.TemporaryDirectory(dir=benchmark.config.scratchdir) as tempdir:
+
+        os.makedirs(os.path.join(tempdir, "upper"))
+        os.makedirs(os.path.join(tempdir, "work"))
+
+        tool_command = " ".join(args)
+        singularity_command = (
+            f"singularity exec "
+            f'-B "$PWD":/lower --no-home '
+            f'-B "{tempdir}":/overlay '
+            f'--fusemount "container:fuse-overlayfs -o lowerdir=/lower -o upperdir=/overlay/upper -o workdir=/overlay/work \\"$HOME\\"" '
+            f"{benchmark.config.singularity} {tool_command}"
+            if benchmark.config.singularity
+            else tool_command
+        )
+        srun_command = (
+            f"srun "
+            f"-t {srun_timelimit} "
+            f"-c {cpus} "
+            f"-o {log_file} "
+            f"--mem-per-cpu {mem_per_cpu} "
+            f"--threads-per-core=1 "
+            f"--ntasks=1 "
+            f"{singularity_command}"
+        )
+        srun_result = subprocess.run(
+            ["bash", "-c", srun_command],
+            shell=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        logging.debug(
+            "srun: returncode: %d, output: %s",
+            srun_result.returncode,
+            srun_result.stdout,
+        )
+        jobid_match = jobid_pattern.search(str(srun_result.stdout))
+        if jobid_match:
+            jobid = int(jobid_match.group(1))
+        else:
+            logging.debug("Jobid not found in stderr, aborting")
+            stop()
+            return -1
+
+        seff_command = f"seff {jobid}"
+        logging.debug("Command to run: %s", seff_command)
+        result = subprocess.run(
+            ["bash", "-c", seff_command],
+            shell=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+
+    # Runexec would populate the first 6 lines with metadata
+    with open(log_file, "r+") as file:
+        content = file.read()
+        file.seek(0, 0)
+        empty_lines = "\n" * 6
+        file.write(empty_lines + content)
+
+    exit_code, cpu_time, wall_time, memory_usage = parse_seff(str(result.stdout))
+
+    return {
+        "walltime": wall_time,
+        "cputime": cpu_time,
+        "memory": memory_usage,
+        "exitcode": ProcessExitCode.create(value=exit_code),
+    }
+
+
+exit_code_pattern = re.compile(r"State: COMPLETED \(exit code (\d+)\)")
+cpu_time_pattern = re.compile(r"CPU Utilized: (\d+):(\d+):(\d+)")
+wall_time_pattern = re.compile(r"Job Wall-clock time: (\d+):(\d+):(\d+)")
+memory_pattern = re.compile(r"Memory Utilized: (\d+\.\d+) MB")
+
+
+def parse_seff(result):
+    logging.debug(f"Got {result} as output from seff")
```

Comment:
Would this potentially be a lot of output? Then better avoid it. It is just a single line, it can be ok, but I would suggest to move the result it to the end of the log message.

And I just noticed the use of f-strings in log messages. That makes the string formatting eager, i.e., also if the log message is not even printed. Using `%` formatting and separate parameters is lazy:
```suggestion
    logging.debug("Seff output: %s", result)
```
Could you please check other log statements as well?

Reply:
This was the only dangerous one, all the others had their potentially large (multiline) outputs at the end.